pkg/provider/provider.go

```diff
@@ -223,105 +219,106 @@ func Provider() *schema.Provider {
 				Optional:      true,
 				Sensitive:     true,
-				DefaultFunc:   schema.EnvDefaultFunc("SNOWFLAKE_PRIVATE_KEY", nil),
+				DefaultFunc:   schema.EnvDefaultFunc(snowflakeenvs.PrivateKey, nil),
 				ConflictsWith: []string{"browser_auth", "password", "oauth_access_token", "private_key_path", "oauth_refresh_token"},
 			},
 			"private_key_passphrase": {
 				Type:          schema.TypeString,
-				Description:   "Supports the encryption ciphers aes-128-cbc, aes-128-gcm, aes-192-cbc, aes-192-gcm, aes-256-cbc, aes-256-gcm, and des-ede3-cbc. Can also be sourced from `SNOWFLAKE_PRIVATE_KEY_PASSPHRASE` environment variable.",
+				Description:   envNameFieldDescription("Supports the encryption ciphers aes-128-cbc, aes-128-gcm, aes-192-cbc, aes-192-gcm, aes-256-cbc, aes-256-gcm, and des-ede3-cbc.", snowflakeenvs.PrivateKeyPassphrase),
 				Optional:      true,
 				Sensitive:     true,
-				DefaultFunc:   schema.EnvDefaultFunc("SNOWFLAKE_PRIVATE_KEY_PASSPHRASE", nil),
+				DefaultFunc:   schema.EnvDefaultFunc(snowflakeenvs.PrivateKeyPassphrase, nil),
 				ConflictsWith: []string{"browser_auth", "password", "oauth_access_token", "oauth_refresh_token"},
 			},
 			"disable_telemetry": {
 				Type:        schema.TypeBool,
-				Description: "Indicates whether to disable telemetry. Can also be sourced from the `SNOWFLAKE_DISABLE_TELEMETRY` environment variable.",
+				Description: envNameFieldDescription("Disables telemetry in the driver.", snowflakeenvs.DisableTelemetry),
 				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_DISABLE_TELEMETRY", nil),
+				DefaultFunc: schema.EnvDefaultFunc(snowflakeenvs.DisableTelemetry, nil),
 			},
 			"client_request_mfa_token": {
-				Type:        schema.TypeBool,
-				Description: "When true the MFA token is cached in the credential manager. True by default in Windows/OSX. False for Linux. Can also be sourced from the `SNOWFLAKE_CLIENT_REQUEST_MFA_TOKEN` environment variable.",
-				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_CLIENT_REQUEST_MFA_TOKEN", nil),
+				Type:             schema.TypeString,
+				Description:      envNameFieldDescription("When true the MFA token is cached in the credential manager. True by default in Windows/OSX. False for Linux.", snowflakeenvs.ClientRequestMfaToken),
+				Optional:         true,
+				DefaultFunc:      schema.EnvDefaultFunc(snowflakeenvs.ClientRequestMfaToken, resources.BooleanDefault),
+				ValidateDiagFunc: validateBooleanString,
 			},
 			"client_store_temporary_credential": {
-				Type:        schema.TypeBool,
-				Description: "When true the ID token is cached in the credential manager. True by default in Windows/OSX. False for Linux. Can also be sourced from the `SNOWFLAKE_CLIENT_STORE_TEMPORARY_CREDENTIAL` environment variable.",
+				Type:        schema.TypeString,
```

Comment:
if the validations works differently for provider and resource configs, is this type change safe for the provider config? (it was safe for the resource)

Reply:
\+ you forgot to add boolean validator here